bpf/bpf_lxc.c

```diff
@@ -30,6 +30,14 @@
 #include "lib/nat46.h"
 #include "lib/identity.h"
 #include "lib/policy.h"
+
+/* Override LB_SELECTION initially defined in node_config.h to force bpf_lxc to use the random backend selection
+ * algorithm for in-cluster traffic. It will fail with the Maglev hash algorithm because Cilium doesn't provision
+ * the Maglev table for ClusterIP.
+ */
```

Comment:
Given ClusterIP can also be accessed from outside per Cilium config, can't we just include ClusterIP for the Maglev LUT? Is there a strong reason to have this special cased? (cc @brb)

Reply:
I thought provisioning the Maglev lookup table for Cluster doesn't make sense, because it will not be accessed from outside cluster unless bpf.lbExternalClusterIP is set to true. However, it might be better to include ClusterIP [here](https://github.com/cilium/cilium/blob/master/pkg/service/service.go#L115) since it would handle both cases where bpf.lbExternalClusterIP is true and false.
What do you think?  @brb

Reply:
My thinking was that populating Maglev LUT for each ClusterIP SVC regardless of the lbExternalClusterIP setting would have a big memory footprint.

Reply:
But if we have lbExternalClusterIP enabled and Maglev as well, isn't that broken then? How is the service table populated/handled for external clients connecting to ClusterIP there?

Reply:
In our cluster,  we have 63 LB services and 242 ClusterIP services. So the number of tables will increase 5-fold if we provision  the Maglev LUT for ClusterIP as well. It might be a waste to do so unconditionally. Do we include ClusterIP for Maglev LUT only if lbExternalClusterIP is set true, while adding `#define LB_SELECTION LB_SELECTION_RANDOM` to bpf_lxc ?

Reply:
That sounds reasonable way forward to me, wdyt @brb?

Reply:
Sounds good to me!